torch/csrc/jit/mobile/interpreter.cpp

```diff
@@ -52,7 +55,10 @@ bool InterpreterState::run(Stack& stack) {
           // enable only for the RecordFunction
           enableRecordFunction(true);
         }
-        RECORD_FUNCTION(code_->op_names_[inst.X].name, stack);
+        std::string fn_name = (!module_debug_info_list_[inst.X].empty()) ?
```

Comment:
I think we should not do this here. String manipulation is critical path will add latency. Given that op_index corresponds to pc, we can add a function in `Function` that constructs a map/vector <pc, debug_string>, probably vector is better and have an API provided at mobile::Module level where you can query debug info for a PC. This can then be used in predictor to augment debug info to op names. This way getting debug info can be optional and if a client does not want debug info, it does not get it. @iseeyuan, what do you think?

Reply:
For this, is it ok to save op names augmented with module info (like `top.A1.aten::add`) in `Function` instead of saving module info (like `top.A1`). This way, we don't have to do string concatenation when running our model. Or should we save both of them in two separate vectors?

Reply:
That is what we wanted to avoid because now interpreter has to reverse engineer to find op name if you append it like that. I think the best thing to do is: After having done append_instruction and append_operator, have a function that iterates over instructions, where each index is PC, find inst[PC].X which is op_index, find corresponding module_debug_info and then instead of having module_debug_info_list_ that has one-to-one correspondence with operator list, make it map<PC, module_debug_info>. Now you can add a function `get_module_debug_info(size_t PC)` which let's you query a Function. This can be exposed via mobile::Module by adding a method `get_forward_method_debug_info(size_t PC)`. So then in predictor when we finally print out op with latency info, we can query debug info using PC which we had also used for mapping op_names.